### PR TITLE
Add docker

### PR DESCRIPTION
**Pull request by alegout**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop.
| Description?  | Add a docker stack to easily have a development environment.
| Type?         | new feature
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no.
| How to test?  | See README.md.

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8601)

### CODE REVIEW

**Comment by alegout** [comment]
*[src/PrestaShopBundle/Install/DatabaseDump.php, line 144 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8601#-L-r1ytGfGvrrI05GsUD:-L-r1pMyWqDnx7OkcJe5:b-426zdp) ([raw file](https://github.com/prestashop/prestashop/blob/a2428711da2252003613ae0775077b2f1897dc68/src/PrestaShopBundle/Install/DatabaseDump.php#L144)):*
> ```PHP
>      * Make a database dump
>      */
>     public static function create($overrideIfExists = true)
> ```

Param has to be deleted

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8601)*

File: tools/docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          - PHP_DISPLAY_ERRORS=1
+          - PHP_DEBUGGER=xdebug
+          - _PS_MODE_DEV_=true
+        depends_on:
+          - mysql
+    mysql:
+        image: mysql
+        container_name: mysql-prestashop
+        volumes:
+            - ./data/mysql:/var/lib/mysql
```
Comment (Quetzacoalt91) [inline tools/docker/docker-compose.yml:27]: Do you really want to store the files in the host filesystem? A volume container wouldn't be more suitable?

File: tools/docker/conf/Apache-Php/DockerFile-Apache-Php [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FROM webdevops/php-apache-dev
+COPY conf/Apache-Php/php.ini /opt/docker/etc/php/php.ini
+COPY conf/Apache-Php/.bash_aliases /root
+RUN apt-get update && \
+    apt-get install -y mysql-client
```
Comment (Quetzacoalt91) [inline tools/docker/conf/Apache-Php/DockerFile-Apache-Php:5]: The mysql client is already available in your `mysql` container, do you really need to install here as well?

The short answer should be no, as it is not used by the PHP / Apache server.
In all cases, FYI you should keep your layers as small as possible. When you run `apt update`, some files are generated, but they are probably not useful for the next next steps. So a cleanup could be nice here, i.e with `&& rm -rf /var/lib/apt/lists/*`

**Review by Quetzacoalt91 — COMMENTED** [review]
Nice job @alegout 

We need to discuss together in order to reuse (and probably adapt) the existing stuff.
* [Releases of PrestaShop](https://hub.docker.com/r/prestashop/prestashop/) - https://github.com/PrestaShop/docker
* [PrestaShop on Git](https://hub.docker.com/r/prestashop/prestashop-git/) - https://github.com/PrestaShop/docker-ci
* Docker template with SQL dump - https://github.com/PrestaShop/docker-templates/tree/master/from-dump
* Mine: https://github.com/PrestaShop/PrestaShop/pull/7721 :grin:

File: tools/docker/conf/Apache-Php/DockerFile-Apache-Php (hunk 2 above)
Comment (alegout) [reply tools/docker/conf/Apache-Php/DockerFile-Apache-Php:5]: mysql-client was needed to launch phpunit tests in CLI, it uses mysqldump.

**Comment by alegout** [comment]
abandoned for #7721 thank you @Quetzacoalt91 for your help and explanations.

**Pull request closed without merging**